src/mongoc/mc_topology.erl

```diff
@@ -128,14 +129,18 @@ get_pool(RPMode, RPTags, State) ->
       {error, timeout}
   end.
 
-get_pool(From, #topology_state{self = Topology, get_pool_timeout = TM} = State, RPMode, Tags) ->
```

Comment:
To support a smooth hot patch, this clause should be kept, the old call might have received already in the mailbox.

Reply:
By the time the module enters the new version